src/Microsoft.Health.ControlPlane.Core/Features/Persistence/IControlPlaneDataStore.cs

```diff
@@ -14,5 +15,15 @@ public interface IControlPlaneDataStore
         Task<IdentityProvider> GetIdentityProviderAsync(string name, CancellationToken cancellationToken);
 
         Task<IdentityProvider> UpsertIdentityProviderAsync(IdentityProvider identityProvider, CancellationToken cancellationToken);
+
+        Task<Role> GetRoleAsync(string name, CancellationToken cancellationToken);
+
+        Task<Role> UpsertRoleAsync(Role role, CancellationToken cancellationToken);
+
+        Task<IEnumerable<Role>> GetRoleAllAsync(CancellationToken cancellationToken);
+
+        Task<string> DeleteRoleAsync(string name, CancellationToken cancellationToken);
+
+        Task HardDeleteRoleAsync(string name, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
We should change the name of this method to `DeleteRoleAsync` and remove the method above. Also, we shouldn't pass in a default value for cancellation token here.